iceoryx2/tests/service_tests.rs

```diff
@@ -24,8 +34,90 @@ mod service {
         .unwrap()
     }
 
+    trait SutFactory {
+        type Factory;
+        type CreateError: std::fmt::Debug;
+        type OpenError: std::fmt::Debug;
+
+        fn create(service_name: &ServiceName) -> Result<Self::Factory, Self::CreateError>;
+        fn open(service_name: &ServiceName) -> Result<Self::Factory, Self::OpenError>;
+
+        fn assert_create_error(error: Self::CreateError);
+        fn assert_open_error(error: Self::OpenError);
+    }
+
+    impl<Sut: Service> SutFactory for publish_subscribe::PortFactory<Sut, u64> {
+        type Factory = publish_subscribe::PortFactory<Sut, u64>;
+        type CreateError = PublishSubscribeCreateError;
+        type OpenError = PublishSubscribeOpenError;
+
+        fn create(service_name: &ServiceName) -> Result<Self::Factory, Self::CreateError> {
+            Sut::new(&service_name).publish_subscribe().create::<u64>()
+        }
+
+        fn open(service_name: &ServiceName) -> Result<Self::Factory, Self::OpenError> {
+            Sut::new(&service_name).publish_subscribe().open::<u64>()
+        }
+
+        fn assert_create_error(error: Self::CreateError) {
+            assert_that!(
+                error,
+                any_of([
+                    PublishSubscribeCreateError::AlreadyExists,
+                    PublishSubscribeCreateError::IsBeingCreatedByAnotherInstance,
+                ])
+            );
```

Comment:
Fun fact. Using braces for the content after `any_of` or `eq` made rustfmt work on the macro